src/test/java/com/amazon/opendistroforelasticsearch/security/ssl/util/CertFromFileTests.java

```diff
@@ -0,0 +1,57 @@
+package com.amazon.opendistroforelasticsearch.security.ssl.util;
+
+import com.amazon.opendistroforelasticsearch.security.test.helper.file.FileHelper;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class CertFromFileTests {
+
+    @Test
+    public void testLoadSameCertForClientServerUsage() throws Exception {
+      CertFileProps certProps = new CertFileProps(
+          FileHelper.getAbsoluteFilePathFromClassPath("ssl/node-0.crt.pem").toString(),
+          FileHelper.getAbsoluteFilePathFromClassPath("ssl/node-0.key.pem").toString(),
+          FileHelper.getAbsoluteFilePathFromClassPath("ssl/root-ca.pem").toString(),
+          null);
+
+      CertFromFile cert = new CertFromFile(certProps);
+
+      Assert.assertEquals(1, cert.getCerts().length);
+      Assert.assertNotNull(cert.getClientPemCert());
+      Assert.assertNotNull(cert.getClientPemKey());
+      Assert.assertNotNull(cert.getClientTrustedCas());
+    }
+
+  @Test
+  public void testLoadCertWithoutCA() throws Exception {
+    CertFileProps certProps = new CertFileProps(
+        FileHelper.getAbsoluteFilePathFromClassPath("ssl/node-0.crt.pem").toString(),
+        FileHelper.getAbsoluteFilePathFromClassPath("ssl/node-0.key.pem").toString(),
+        null,
+        null);
+
+    CertFromFile cert = new CertFromFile(certProps);
+
+    Assert.assertNull(cert.getClientTrustedCas());
+  }
+
```

Comment:
Can you also add one negative test to verify if extended key is enabled but either server/client cert file or both are missing ?